lib/pretty_text.rb

```diff
@@ -348,6 +352,25 @@ def self.strip_hidden_unicode_bidirectional_characters(doc)
     end
   end
 
+  def self.sanitize_hotlinked_media(doc)
+    return if !SiteSetting.block_hotlinked_media
+
+    allowed_pattern = allowed_src_pattern
+
+    doc.css("img[src], source[src], source[srcset], track[src]").each do |el|
+      if el["src"] && !el["src"].match?(allowed_pattern)
+        el[PrettyText::BLOCKED_HOTLINKED_SRC_ATTR] = el.delete("src")
+      end
+
+      if el["srcset"]
+        srcs = el["srcset"]&.split(',')&.map { |e| e.split(' ', 2)[0].presence }
+        if srcs.any? { |src| !src.match?(allowed_pattern) }
```

Comment:
```suggestion
        if srcs.all? { |src| src.match?(allowed_pattern) }
```
Minor suggestion to avoid the negative expression.

Reply:
The original code allows for an early return, so the negative expression makes sense here.